<launch>
  <arg name="use_sim_time" default="true" />
  <arg name="gui" default="true" />
  <arg name="headless" default="false" />
  <arg name="world_name" default="$(find gazebo_radiation_plugins)/custom_models/SI/si.world" />

  <param name="use_sim_time" value="$(arg use_sim_time)"/>

  <!-- Short-term hack to support the original front_laser:=true argument for spawning
       the simulator. This will be removed in favour of using the config:=x arg instead. -->
  <arg name="front_laser" default="true" />
  <arg name="default_config" value="front_laser" if="$(arg front_laser)" />
  <arg name="default_config" value="base" unless="$(arg front_laser)" /> 
  <!-- end of hack -->

  <!--Load params -->
  <rosparam file="$(find gazebo_radiation_plugins)/custom_models/SI/configs/attentuation.yaml" command="load"  />
  <rosparam file="$(find gazebo_radiation_plugins)/custom_models/SI/configs/sensors.yaml" command="load"  />
  <rosparam file="$(find gazebo_radiation_plugins)/custom_models/SI/configs/radiation.yaml" command="load"  />

  <!-- Launch Gazebo with the specified world -->

  <node name="gazebo" pkg="gazebo_radiation_plugins" type="gazebo"  output="screen"
  args="--verbose $(arg world_name)" />

  <!-- Configuration of Jackal which you would like to simulate.
       See jackal_description for details. -->
       <arg name="config" default="$(arg default_config)" />

  <!-- Attach sensor to Jackal -->
  <node pkg="tf" type="static_transform_publisher" name="sim_sensor_to_robot" args="0 0 0 0 0 0 base_link sim_sensor 100"/>

  <!-- Run gmapping -->
  <!-- include file="$(find jackal_navigation)/launch/include/gmapping.launch" /-->

  <!-- Run Move Base -->
  <!-- include file="$(find gazebo_radiation_plugins)/launch/move_base.launch" /-->

  <!-- <include file="$(find gazebo_radiation_plugins)/launch/radmap.launch" /> -->

  <!-- Run RVIZ -->
  <node name="rviz" pkg="rviz" type="rviz"  args="-d $(find gazebo_radiation_plugins)launch/rviz_config/husky_radiation_mapping.rviz"/>

</launch>
